File: Two Pointers/3239.Minimum Number of Flips to Make Binary Grid Palindromic I/Solution.java
class Solution {
    public int minFlips(int[][] grid) {
        int m = grid.length, n = grid[0].length;
        int ans1 = 0;
        for (int i = 0; i < m; i++) {
            int l = 0, r = n - 1;
            while (l < r) {
                if (grid[i][l++] != grid[i][r--]) {
                    ans1++;
                }
            }
        }
        int ans2 = 0;
        for (int i = 0; i < n; i++) {
            int l = 0, r = m - 1;
            while (l < r) {
                if (grid[l++][i] != grid[r--][i]) {
                    ans2++;
                }
            }
        }
        return Math.min(ans1, ans2);
    }
}

File: Two Pointers/3239.Minimum Number of Flips to Make Binary Grid Palindromic I/Solution.py
class Solution:
    def minFlips(self, grid: List[List[int]]) -> int:
        m, n = len(grid), len(grid[0])
        ans1 = 0
        for i in range(m):
            l, r = 0, n - 1
            while l < r:
                if grid[i][l] != grid[i][r]:
                    ans1 += 1
                l += 1
                r -= 1

        ans2 = 0
        for i in range(n):
            l, r = 0, m - 1
            while l < r:
                if grid[l][i] != grid[r][i]:
                    ans2 += 1
                l += 1
                r -= 1
        return min(ans1, ans2)
